src/cuda/image_function_cuda.cu

```diff
@@ -43,6 +44,16 @@ namespace
     const FunctionRegistrator functionRegistrator;
 
     // The list of CUDA device functions on device side
+    __global__ void normalizeCuda( const uint8_t * image, uint32_t rowSize, bool horizontal, uint32_t width, uint32_t height, uint32_t * projection )
+    {
+        const uint32_t x = blockDim.x * blockIdx.x + threadIdx.x;
+        const uint32_t y = blockDim.y * blockIdx.y + threadIdx.y;
+
+        if ( x < width && y < height ) {
```

Comment:
This function is not completed.

Reply:
Should I refer to the OpenCL implementation on `projectionHorizontalOpenCL` and `projectionVerticalOpenCL` based on the `horizontal` value?

Reply:
Yes.